File: leetcode-java/src/test/java/com/wezhyn/learn/array/MajorityElementTest.java
package com.wezhyn.learn.array;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * @author wezhyn
 * @since 08.29.2020
 */
public class MajorityElementTest {

    @Test
    public void majorityElement() {
        assertEquals(2, new MajorityElement().majorityElement(new int[]{2, 2, 1, 1, 1, 2, 2}));
        assertEquals(3, new MajorityElement().majorityElement(new int[]{3, 5, 3}));
        assertEquals(3, new MajorityElement().majorityElement(new int[]{3, 2, 3}));
    }
}

File: leetcode-kotlin/src/test/kotlin/com/wezhyn/learn/lcof/MajorityElementTest.kt
package com.wezhyn.learn.lcof

import com.wezhyn.learn.intArray
import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * @author wezhyn
 * @since 07.02.2020
 */
class MajorityElementTest {

    @Test
    fun majorityElement() {
        assertEquals(3, MajorityElement().majorityElement("""3,2,3""".intArray()))
        assertEquals(4, MajorityElement().majorityElement("4,5,4".intArray()))
        assertEquals(2, MajorityElement().majorityElement("2,2,1,1,1,2,2".intArray()))
    }
}